server/server.go

```diff
@@ -387,21 +388,50 @@ func (s *Server) KillAllConnections() {
 	}
 }
 
+var gracefulCloseConnectionsTimeout = 15 * time.Second
+
+// TryGracefulDown will try graceful close all connection first with timeout. if timeout, will close all connection directly.
+func (s *Server) TryGracefulDown() {
+	ctx, cancel := context.WithCancel(context.Background())
+	timeout := time.Tick(gracefulCloseConnectionsTimeout)
+	ch := make(chan struct{})
+	go func() {
+		s.GracefulDown(ctx)
+		close(ch)
+	}()
+	select {
+	case <-timeout:
+		cancel()
+		s.KillAllConnections()
```

Comment:
There is a race condition that `<-timeout` and `s.GracefulDown(ctx)` both triggered.
Will the connection be close twice ?